File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prod = 1
        count = 0
        for n in nums:
            if n != 0:
                prod *= n
            else:
                count += 1

        ret = []

        for n in nums:
            if count > 1:
                ret.append(0)

            elif n == 0 and count == 1:
                ret.append(prod)
            elif n!= 0 and count ==1:
                ret.append(0)
            else:
                ret.append(prod//n)

        return ret
        

File: Data Structures & Algorithms/subsets/submission-0.cpp
class Solution {
public:
    vector<vector<int>> ret;
    void bfs(vector<int>& nums, vector<int>& sublist, int idx) {
        if (idx == nums.size()) {
            ret.push_back(sublist);
            return;
        }

        sublist.push_back(nums[idx]);

        bfs(nums, sublist, idx+1);
        sublist.pop_back();
        bfs(nums, sublist, idx+1);
    } 
    
    vector<vector<int>> subsets(vector<int>& nums) {

        vector<int> sublist;
        bfs(nums, sublist, 0);

        return ret;
    }
};
